// containers/login/index.js
import React, { Component } from 'react';
import Button from 'components/button';
import FormInput from 'components/formInput';
import { connect } from 'react-redux';
import { loginUser } from 'actions';
import R from 'ramda';

class Login extends Component {
    constructor(props) {
        super(props);
        this.loginValue = '';
        this.passwordValue = '';
    }
    componentWillMount() {
        if(this.props.userInfo.name) {
            this.props.router.push('/')
        }
    }
    onLoginChange (value) {
      this.loginValue = value
    };
    onPasswordChange (value) {
        this.passwordValue = value
    };
    async logInput () {
      console.log('loginValue', this.loginValue);
      console.log('passwordValue', this.passwordValue);
      await this.props.loginUser(this.loginValue, this.passwordValue);
      this.props.router.push('/');
    };
    render() {
        return (
            <div>
                LOGIN
                <FormInput onChange={this.onPasswordChange.bind(this)}/>
                <FormInput onChange={this.onLoginChange.bind(this)}/>
                <Button handler={this.logInput.bind(this)} name={`Login`}/>
            </div>
        )
    }
};
const mapStateToProps = (state) => ({
    userInfo: state.user
});
const mapDispatchToProps = {
    loginUser
};
export default connect(mapStateToProps, mapDispatchToProps)(Login);
